### PR TITLE
Upgrading multilingual site causes DB Error

### PR DESCRIPTION
**Pull request by chamilwijesooriya**
[GitLab Issue 311](https://lab.civicrm.org/dev/core/issues/311) - Upgrading multilingual site causes DB Error

Overview
----------------------------------------
Upgrading from CiviCRM 4.4.7 to 5.4.0 (Drupal), cause a DB error. Upgrade process works fine.

Before
----------------------------------------
When loading any civi page, JavaScript error is displayed on the language file (backtrace displayed after enabling).

javascript

Checking the error log, few fields relating to uf_group are not getting created during the upgrade.

errorlog

Further investigation showed that in CRM/Core/I18n, there have been schema changes between the latest SchemaStructure.php and SchemaStructure_4_7_alpha1.php

schemastructure

After
----------------------------------------
Copied  CRM/Core/I18n/SchemaStructure.php and renamed to SchemaStructure_5_4_alpha1.php

### CODE REVIEW

File: CRM/Core/I18n/SchemaStructure_5_4_alpha1.php [hunk 1]
```diff
@@ -0,0 +1,714 @@
...
+ */
+class CRM_Core_I18n_SchemaStructure_5_4_alpha1 {
+
+  /**
+   * Get translatable columns.
+   *
+   * @return array
+   *   A table-indexed array of translatable columns.
+   */
+  public static function &columns() {
```
Comment (eileenmcnaughton) [inline CRM/Core/I18n/SchemaStructure_5_4_alpha1.php]: Just as a general pattern - we have been removing the & in lines like above for a while now. It's there in old code so I understand you copying it in - but we don't think it's a good thing

**Comment by eileenmcnaughton** [comment]
How does this work?

Note that in 4.7.31 a new multilingual field was added - for multilingual upgraders running the api 
System.rebuildmultilingualschema should work - not ideal but....

@seamuslee001 @mlutfy thoughts

**Comment by mlutfy** [comment]
This would explain why we need to run `rebuildmultilingualschema` after upgrades. It's possible that the new DB Schema does not get loaded by the upgrader, so when it rebuilds the schema, it does not include the `frontend_title` in the view.

It might have been better to name it `SchemaStructure_4_7_31.php`, but since we don't support old versions anyway, this isn't a big deal.

**Comment by chamilwijesooriya** [comment]
@eileenmcnaughton oh didn't know about the API call.. couldn't find in the docs 

I just followed what's mentioned [in the docs here](https://docs.civicrm.org/dev/en/latest/translation/database/#localised-fields-schema-changes) 

docs

and I guess this is the code where it choses which schema structure to load in CRM/Core/I18n/Schema.php

code

**Comment by eileenmcnaughton** [comment]
@seamuslee001 I think this is mergeable ....

**Comment by mlutfy** [comment]
Seems OK to me as well

**Comment by seamuslee001** [comment]
Seems sensible to me, I say we merge it

**Comment by colemanw** [comment]
If it's a regression, should it go in 5.5?

**Comment by eileenmcnaughton** [comment]
@colemanw it's not a recent regression - it has affected the whole 5.x series as it was 4.7.31

**Pull request merged**